src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepExecution.java

```diff
@@ -102,29 +103,26 @@ protected Job asJob() {
                 throw new AbortException("Item type does not support parameters");
             }
             Queue.Task task = (Queue.Task) item;
+            listener.getLogger().println("Scheduling item: " + ModelHyperlinkNote.encodeTo(item));
             node.addAction(new LabelAction(Messages.BuildTriggerStepExecution_building_(task.getFullDisplayName())));
-            List<Action> actions = new ArrayList<>();
-            if (step.getWait()) {
-                StepContext context = getContext();
-                actions.add(new BuildTriggerAction(context, step.isPropagate()));
```

Comment:
Well. There could be a `Job & Queue.Task` which is not a `ParameterizedJobMixIn.ParameterizedJob`. I do not know of any examples (`ExternalJob` is not schedulable) so this is hypothetical, but we could wind up in a situation where we accept `wait` and such a job, and so the step starts and goes to sleep but then will never wake up on its own.

(Yes I know I said you could use `singletonList`; I forgot about the `BuildTriggerAction`.)